packages/dds/map/src/test/directory.spec.ts

```diff
@@ -1036,5 +1036,147 @@ describe("Directory", () => {
                 assert.ok(expectedDirectories2.size === 0);
             });
         });
+
+        describe("Garbage Collection", () => {
+            it("can generate GC nodes with handles in data", () => {
+                const subMap = mapFactory.create(dataStoreRuntime, "subMap");
+                directory.set("object", subMap.handle);
+
+                const fooDirectory = directory.createSubDirectory("foo");
+                const subMap2 = mapFactory.create(dataStoreRuntime, "subMap2");
+                fooDirectory.set("object", subMap2.handle);
+
+                const barDirectory = fooDirectory.createSubDirectory("bar");
+                const subMap3 = mapFactory.create(dataStoreRuntime, "subMap3");
+                barDirectory.set("object", subMap3.handle);
+
+                const bazDirectory = barDirectory.createSubDirectory("baz");
+                const subMap4 = mapFactory.create(dataStoreRuntime, "subMap4");
+                bazDirectory.set("object", subMap4.handle);
+
+                containerRuntimeFactory.processAllMessages();
+
+                // Verify the GC nodes returned by summarize.
+                const gcNodes = directory2.summarize().gcNodes;
+                assert.strictEqual(gcNodes.length, 1, "There should only be one GC node in summary");
+                assert.strictEqual(gcNodes[0].id, "/", "GC node's id should be /");
+                assert.deepStrictEqual(
+                    gcNodes[0].outboundRoutes,
+                    [
+                        subMap.handle.absolutePath,
```

Comment:
Nit: order is not important here (from API perspective), but is important for test to pass.
We have same problem in summarization tests. Should we do something about it? :)

Reply:
You right! We might end up in similar problems. We should definitely do something about it. I will fix this :-).

Reply:
Sorted the arrays before comparing.